### PR TITLE
Session 2 - Storage

### PR DESCRIPTION
**Pull request by sgibson91**
After a chat with @DavidBeavan on Friday and reflecting on how much time we had last week, I'm keeping this session much shorter. We can move on to the materials in #8 if we have time left over.

### CODE REVIEW

File: sessions/02_Storage.rst [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+In this session, we're going to look at the different types of storage available in Azure and when to use them.
+
+Introduction
+============
+
+In Azure, there's much more available than "just" somewhere to store files or virtual disks for your VMs.
+
+Let's think about a fictional pizza company running an online service!
+The "backend" app needs a data store that holds the available pizzas, list of toppings, and prices.
```
Comment (DavidBeavan) [inline sessions/02_Storage.rst]: Might need some extra on back- front-end terminology

File: sessions/02_Storage.rst [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+-------------
+
+You may have heard of SQL databases such as Microsoft SQL Server, MySQL or PostgreSQL.
+These are *relational* databases, made up of one or more tables that contain one or more rows of data.
+These types of databases are common for application development and can be designed, visualised, and queried in a structured manner - the *S* in *SQL* (for Structured Query Language).
+
+NoSQL databases are different.
+They don't follow the same structured approach, and data isn't stored in tables where each row contains the same fields.
+There are different implementations of NoSQL databases such as: MongoDB, Cassandra, SAP HANA, CouchDB, and Redis.
+The advantages of NoSQL databases are that they scale horizontally (you can add more servers rather than more memory or CPU), can handle larger amounts of data, and are more efficient at processing those large datasets.
```
Comment (DavidBeavan) [inline sessions/02_Storage.rst]: But the downsides? NoSQL systems can easily enter inconsistent states. The query languages and APIs are not standardised like SQL, locking you in. They are less mature and have smaller communities of support. Discuss.

File: sessions/02_Storage.rst [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+* *Key-value* (like a dictionary), such as Redis,
+* *Column*, such as Cassandra,
+* *Document*, such as MongoDB.
+
+Each approach has pros and cons for performance, flexibility, or complexity point of view.
+An Azure storage table uses a *key-value* and is a good introduction to NoSQL databases.
+
++----------------------------------------------------------------------------------------------------------------+
+| **Info:** You can download and install the Microsoft Azure Storage Explorer if you like to visualise the data. |
```
Comment (DavidBeavan) [inline sessions/02_Storage.rst]: not just visualise, but edit content and amend the schema etc.

File: sessions/02_Storage.rst [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+| www.storageexplorer.com                                                                                        |
++----------------------------------------------------------------------------------------------------------------+
+
+Exercise
+--------
+
+We are now going to see Azure Tables in action using Python!
+
+1. Open the Azure Portal in a web browser and then open the Cloud Shell.
```
Comment (DavidBeavan) [inline sessions/02_Storage.rst]: What was the cloud shell-only link?

File: sessions/02_Storage.rst (hunk 1 above)
Comment (sgibson91) [reply sessions/02_Storage.rst]: [`089cd65`](https://github.com/alan-turing-institute/learn-azure/pull/7/commits/089cd65d168fbcb780ed90f7b5a8670aaaf9fc93)

File: sessions/02_Storage.rst (hunk 2 above)
Comment (sgibson91) [reply sessions/02_Storage.rst]: [`80b4cc0`](https://github.com/alan-turing-institute/learn-azure/pull/7/commits/80b4cc05528d9926f62af59d6bf580bda65f7e62)

File: sessions/02_Storage.rst (hunk 4 above)
Comment (sgibson91) [reply sessions/02_Storage.rst]: shell.azure.com

File: sessions/02_Storage.rst (hunk 3 above)
Comment (sgibson91) [reply sessions/02_Storage.rst]: [`4552431`](https://github.com/alan-turing-institute/learn-azure/pull/7/commits/4552431c47246188d07ba872e12bb96b8f9bf742)

**Pull request merged**